### PR TITLE
[SPARK-10100] [SQL] Perfomance improvements to new MIN/MAX aggregate functions.

### PR DESCRIPTION
**Pull request by hvanhovell**
The new MIN/MAX suffer from a performance regression. This PR aims to fix this by simplifying the evaluation of the MIN/MAX functions.

See the JIRA [ticket](https://issues.apache.org/jira/browse/SPARK-10100) for more information.

### CODE REVIEW

**Comment by yhuai** [comment]
ok to test

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/functions.scala [hunk 1]
```diff
@@ -236,15 +234,13 @@ case class Min(child: Expression) extends AlgebraicAggregate {
   )
 
   override val updateExpressions = Seq(
-    /* min = */ If(IsNull(child), min, If(IsNull(min), child, Least(Seq(min, child))))
+    /* min = */  If(IsNull(min), child, If(GreaterThan(min, child), child, min))
```
Comment (adrian-wang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/functions.scala]: Nit: redundant space.

**Comment by adrian-wang** [comment]
I doubt that whether your decision tree is better. But GreaterThan/LessThan should be a little better than Least/Greatest, I think.

**Comment by hvanhovell** [comment]
@adrian-wang the improvement is absolutely tiny, about 2-3% if you do a lot of ```min```'s of ```max```'es.

This PR was a response to misdiagnosed performance regression, the real cause was the use of a map in key-less aggregation. The PR adds some value, we could add it to 1.6. @yhuai / @adrian-wang if you feel differently about this, I'll withdraw the PR.

**Pull request closed without merging**